netty/src/main/java/io/grpc/netty/GrpcSslContexts.java

```diff
@@ -225,6 +226,18 @@ public static SslContextBuilder configure(SslContextBuilder builder, Provider jd
         throw new IllegalArgumentException(
             SUN_PROVIDER_NAME + " selected, but Jetty NPN/ALPN unavailable");
       }
+    } else if (IBM_PROVIDER_NAME.equals(jdkProvider.getName())) {
+      // Jetty ALPN/NPN only supports one of NPN or ALPN
+      if (JettyTlsUtil.isJettyAlpnConfigured()) {
```

Comment:
Does it actually support Jetty ALPN? I don't see anything about Jetty ALPN working for IBM and based on the [IBM documentation](https://www.ibm.com/support/knowledgecenter/en/SSEQTP_liberty/com.ibm.websphere.wlp.doc/ae/cwlp_alpnsupport.html), it seems like Jetty ALPN only works if using Oracle's JDK on IBM hardware. It looks like it has [its own API for Java 8](https://www.ibm.com/support/knowledgecenter/en/SSYKE2_8.0.0/com.ibm.java.security.api.doc/jsse2/com/ibm/jsse2/ext/ALPNJSSEExt.html).

I would _really_ like to know which of these is being selected when it works for you. I hope it is the `isJava9AlpnAvailable()` case.

My eventual goal in this part of the code was to use the Java 9+ API, when available, and check whether this specific provider supports ALPN by creating an sslEngine and [calling `getApplicationProtocol()`](https://docs.oracle.com/javase/9/docs/api/javax/net/ssl/SSLEngine.html#getApplicationProtocol--) (or similar); if it throws UnsupportedOperationException, then it doesn't support ALPN. That way we don't have to hard-code provider names.

SunJSSE is hard-coded today because it _must_ be hard-coded for Jetty ALPN/NPN. Java 9+ support was a community contribution and just followed most of the current flow. Unfortunately, just being on Java 9+ doesn't mean the provider supports ALPN, thus the need for eventually adding a `getApplicationProtocol()` check. But this had been "good enough" up until now since we know Java 9+'s SunJSSE implementation supports ALPN.